Pybuilder does not work with pip 10 beta

Using pybuilder 0.11.15 fails with the latest pip beta release. Dont know whether the pybuilder prerelease version is affected as well.

```
13:48:40 + pyb install_dependencies
13:48:40 Traceback (most recent call last):
13:48:40   File "/xxx/venv/bin/pyb", line 21, in <module>
13:48:40     import pybuilder.cli
13:48:40   File "/xxx/venv/local/lib/python2.7/site-packages/pybuilder/cli.py", line 31, in <module>
13:48:40     from pybuilder.core import Logger
13:48:40   File "/xxx/venv/local/lib/python2.7/site-packages/pybuilder/core.py", line 35, in <module>
13:48:40     from pybuilder import pip_common
13:48:40   File "/xxx/venv/local/lib/python2.7/site-packages/pybuilder/pip_common.py", line 21, in <module>
13:48:40     from pip.commands.show import search_packages_info
13:48:40 ImportError: No module named commands.show
```

The following warning might be helpful as well:
```
13:48:39 pybuilder 0.11.15 has requirement pip>=7.1, but you'll have pip 10.0.0b2 which is incompatible.
```

I'm getting this error running the new build:

```
(venv) ubuntu@ip-10-0-0-102:~/Projects/etl$ pyb -X install_dependencies
PyBuilder version 0.12.0.dev20180411184223
Build started at 2018-04-12 12:05:11
------------------------------------------------------------
[DEBUG] Loading project module from /home/ubuntu/Projects/etl/build.py
[DEBUG] Loading plugin 'python.core'
[DEBUG] Trying to load builtin plugin 'python.core'
[DEBUG] Loading plugin 'core'
[DEBUG] Trying to load builtin plugin 'core'
[DEBUG] Found builtin plugin 'pybuilder.plugins.core_plugin'
[DEBUG] Found task 'clean' with dependencies []
[DEBUG] Registering task 'clean'
[DEBUG] Found task 'compile_sources' with dependencies [prepare]
[DEBUG] Registering task 'compile_sources'
[DEBUG] Registering initializer 'init'
[DEBUG] Found task 'install' with dependencies [package, publish(optional)]
[DEBUG] Registering task 'install'
[DEBUG] Found task 'package' with dependencies [compile_sources, run_unit_tests(optional)]
[DEBUG] Registering task 'package'
[DEBUG] Found task 'prepare' with dependencies []
[DEBUG] Registering task 'prepare'
[DEBUG] Found task 'print_module_path' with dependencies []
[DEBUG] Registering task 'print_module_path'
[DEBUG] Found task 'print_scripts_path' with dependencies []
[DEBUG] Registering task 'print_scripts_path'
[DEBUG] Found task 'publish' with dependencies [package, verify(optional)]
[DEBUG] Registering task 'publish'
[DEBUG] Found task 'run_integration_tests' with dependencies [package]
[DEBUG] Registering task 'run_integration_tests'
[DEBUG] Found task 'run_unit_tests' with dependencies [compile_sources]
[DEBUG] Registering task 'run_unit_tests'
[DEBUG] Found task 'verify' with dependencies [run_integration_tests]
[DEBUG] Registering task 'verify'
[DEBUG] Found builtin plugin 'pybuilder.plugins.python.core_plugin'
[DEBUG] Registering initializer 'init_python_directories'
[DEBUG] Found task 'package' with dependencies []
[DEBUG] Registering task 'package'
[DEBUG] Loading plugin 'python.unittest'
[DEBUG] Trying to load builtin plugin 'python.unittest'
[DEBUG] Found builtin plugin 'pybuilder.plugins.python.unittest_plugin'
[DEBUG] Registering initializer 'init_test_source_directory'
[DEBUG] Found task 'run_unit_tests' with dependencies []
[DEBUG] Registering task 'run_unit_tests'
[DEBUG] Loading plugin 'python.install_dependencies'
[DEBUG] Trying to load builtin plugin 'python.install_dependencies'
[DEBUG] Found builtin plugin 'carrotetl.core.build.pyb.install_dependencies_plugin'
[DEBUG] Found action create_install_log_directory
[DEBUG] Registering action 'create_install_log_directory'
[DEBUG] Registering initializer 'initialize_install_dependencies_plugin'
[DEBUG] Found task 'install_build_dependencies' with dependencies []
[DEBUG] Registering task 'install_build_dependencies'
[DEBUG] Found task 'install_dependencies' with dependencies []
[DEBUG] Registering task 'install_dependencies'
[DEBUG] Found task 'install_runtime_dependencies' with dependencies []
[DEBUG] Registering task 'install_runtime_dependencies'
[DEBUG] Found task 'list_dependencies' with dependencies []
[DEBUG] Registering task 'list_dependencies'
[DEBUG] Have loaded plugins python.core, core, python.unittest, python.install_dependencies
[DEBUG] Found task 'docker_build' with dependencies []
[DEBUG] Registering task 'docker_build'
[DEBUG] Found task 'docker_build_remotely' with dependencies []
[DEBUG] Registering task 'docker_build_remotely'
[DEBUG] Found task 'docker_exec' with dependencies []
[DEBUG] Registering task 'docker_exec'
[DEBUG] Found task 'docker_run' with dependencies []
[DEBUG] Registering task 'docker_run'
[DEBUG] Found task 'docker_run_debugger_prepare' with dependencies []
[DEBUG] Registering task 'docker_run_debugger_prepare'
[DEBUG] Found task 'docker_run_remotely' with dependencies []
[DEBUG] Registering task 'docker_run_remotely'
[DEBUG] Found task 'docker_tag_push' with dependencies []
[DEBUG] Registering task 'docker_tag_push'
[DEBUG] Found task 'docker_up' with dependencies []
[DEBUG] Registering task 'docker_up'
[DEBUG] Found task 'entercontainer' with dependencies []
[DEBUG] Registering task 'entercontainer'
[DEBUG] Found task 'install_r_library' with dependencies []
[DEBUG] Registering task 'install_r_library'
[DEBUG] Found task 'integration_tests' with dependencies []
[DEBUG] Registering task 'integration_tests'
[DEBUG] Found task 'integration_tests_remotely' with dependencies []
[DEBUG] Registering task 'integration_tests_remotely'
[DEBUG] Found task 'push_tmpbuild' with dependencies []
[DEBUG] Registering task 'push_tmpbuild'
[DEBUG] Found task 'register_task_definition_task' with dependencies []
[DEBUG] Registering task 'register_task_definition_task'
[DEBUG] Found task 'save_config_file' with dependencies []
[DEBUG] Registering task 'save_config_file'
[DEBUG] Registering initializer 'set_properties'
[DEBUG] Found task 'update_stack' with dependencies []
[DEBUG] Registering task 'update_stack'
[DEBUG] Adding 'prepare' as a dependency of task 'compile_sources'
[DEBUG] Adding 'package' as a dependency of task 'install'
[DEBUG] Adding 'publish(optional)' as a dependency of task 'install'
[DEBUG] Adding 'compile_sources' as a dependency of task 'package'
[DEBUG] Adding 'run_unit_tests(optional)' as a dependency of task 'package'
[DEBUG] Adding 'package' as a dependency of task 'publish'
[DEBUG] Adding 'verify(optional)' as a dependency of task 'publish'
[DEBUG] Adding 'package' as a dependency of task 'run_integration_tests'
[DEBUG] Adding 'compile_sources' as a dependency of task 'run_unit_tests'
[DEBUG] Adding 'run_integration_tests' as a dependency of task 'verify'
[DEBUG] Adding before action 'create_install_log_directory' for task 'install_build_dependencies'
[DEBUG] Adding before action 'create_install_log_directory' for task 'install_runtime_dependencies'
[DEBUG] Adding before action 'create_install_log_directory' for task 'install_dependencies'
[DEBUG] Verbose output enabled.

[DEBUG] Executing initializer 'init' from 'pybuilder.plugins.core_plugin'
[DEBUG] Executing initializer 'init_python_directories' from 'pybuilder.plugins.python.core_plugin'
[DEBUG] Executing initializer 'init_test_source_directory' from 'pybuilder.plugins.python.unittest_plugin'
[DEBUG] Executing initializer 'initialize_install_dependencies_plugin' from 'carrotetl.core.build.pyb.install_dependencies_plugin'
[DEBUG] Executing initializer 'set_properties' from 'build'
[DEBUG] Project properties:
                                 basedir : /home/ubuntu/Projects/etl
                                dir_dist : $dir_target/dist/carrotetl-7638067
                        dir_dist_scripts : scripts
                        dir_install_logs : $dir_logs/install_dependencies
                                dir_logs : $dir_target/logs
                             dir_reports : $dir_target/reports
                  dir_source_main_python : src/main/python
                 dir_source_main_scripts : src/main/scripts
              dir_source_unittest_python : src/unittest/python
                              dir_target : target
    install_dependencies_extra_index_url : None
         install_dependencies_from_wheel : None
          install_dependencies_index_url : None
install_dependencies_insecure_installation : []
      install_dependencies_local_mapping : {}
       install_dependencies_trusted_host : None
            install_dependencies_upgrade : False
                    unittest_file_suffix : None
                    unittest_module_glob : *_tests
                         unittest_runner : (<function <lambda> at 0x7f2aa68b7d70>, u'_make_result')
             unittest_test_method_prefix : None
                                 verbose : True
[DEBUG] Execution plan is install_dependencies
[INFO]  Building carrotetl version 7638067
[INFO]  Executing build in /home/ubuntu/Projects/etl
[INFO]  Going to execute task install_dependencies
[DEBUG] Executing task 'install_dependencies'
[DEBUG] Executing action 'create_install_log_directory' from 'carrotetl.core.build.pyb.install_dependencies_plugin' before task
[DEBUG] Creating log directory 'target/logs/install_dependencies'
[DEBUG] Executing subtask from carrotetl.core.build.pyb.install_dependencies_plugin
[INFO]  Installing all dependencies
[DEBUG] Inspecting dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6e10>'
[DEBUG] Dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6e10>' is a requirement file and will be included
[DEBUG] Inspecting dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6dd0>'
[DEBUG] Dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6dd0>' is a requirement file and will be included
[DEBUG] Dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6e10>' will be included in batch install
[INFO]  Processing batch dependency 'requirements.build.txt'
[DEBUG] Dependency '<pybuilder.core.RequirementsFile object at 0x7f2aa68b6dd0>' will be included in batch install
[INFO]  Processing batch dependency 'requirements.txt'
Traceback (most recent call last):
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/cli.py", line 413, in main
    environments=options.environments, tasks=arguments)
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/reactor.py", line 128, in build
    return self.build_execution_plan(tasks, execution_plan)
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/reactor.py", line 166, in build_execution_plan
    reactor=self)
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/execution.py", line 351, in execute_execution_plan
    summaries.append(self.execute_task(task, **keyword_arguments))
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/execution.py", line 298, in execute_task
    task.execute(self.logger, keyword_arguments)
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/execution.py", line 169, in execute
    executable.execute(argument_dict)
  File "/home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages/pybuilder/execution.py", line 100, in execute
    self.callable(*arguments)
  File "/home/ubuntu/Projects/etl/src/main/python/carrotetl/core/build/pyb/install_dependencies_plugin.py", line 61, in install_dependencies
    install_dependency(logger, project, as_list(project.build_dependencies) + as_list(project.dependencies))
  File "/home/ubuntu/Projects/etl/src/main/python/carrotetl/core/build/pyb/install_dependencies_plugin.py", line 143, in install_dependency
    log_file=log_file)
  File "/home/ubuntu/Projects/etl/src/main/python/carrotetl/core/build/pyb/install_dependencies_plugin.py", line 221, in _do_install_dependency
    from_wheel=from_wheel)
  File "/home/ubuntu/Projects/etl/src/main/python/carrotetl/core/build/pyb/pip_utils.py", line 62, in pip_install
    from_wheel=from_wheel
  File "/home/ubuntu/Projects/etl/src/main/python/carrotetl/core/build/pyb/pip_utils.py", line 127, in build_pip_install_options
    if constraint_file and pip_common._pip_supports_constraints():
AttributeError: 'module' object has no attribute '_pip_supports_constraints'
------------------------------------------------------------
BUILD FAILED - 'module' object has no attribute '_pip_supports_constraints'
------------------------------------------------------------
Build finished at 2018-04-12 12:05:11
Build took 0 seconds (59 ms)
```
Pip version:
```
(venv) ubuntu@ip-10-0-0-102:~/Projects/etl$ pip --version
pip 9.0.3 from /home/ubuntu/Projects/etl/venv/local/lib/python2.7/site-packages (python 2.7)
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -34,6 +34,9 @@ test_script:
   - ps: .\venv\Scripts\activate.ps1
   - cmd: where python
   - cmd: python --version
-  - cmd: pip install -U setuptools~=36.0
+  - cmd: pip --version
+  - ps: Start-FileDownload 'https://bootstrap.pypa.io/2.6/get-pip.py'
+  - cmd: "python get-pip.py"
+  - cmd: pip --version
   - cmd: python build.py -v -X install_dependencies
   - cmd: python build.py -v -X analyze install

--- a/build.py
+++ b/build.py
@@ -78,7 +78,7 @@ authors = [Author("Alexander Metzner", "alexander.metzner@gmail.com"),
            ]
 url = "http://pybuilder.github.io"
 license = "Apache License"
-version = "0.11.15"
+version = "0.11.16"
 
 requires_python = ">=2.6,!=3.0,!=3.1,!=3.2,<3.7"
 
@@ -97,7 +97,8 @@ def initialize(project):
         project.build_depends_on("importlib")  # for fluentmock
 
     project.depends_on("tblib")
-    project.depends_on("pip", ">=7.1")
+
+    project.depends_on("pip", ">=7.1,!=10.*,<10dev")
     project.depends_on("setuptools", "~=36.0")
     project.depends_on("wheel", "~=0.29.0")
     project.depends_on("tailer")

